File: code_eval/java/easy/Challenge04.java
package easy;

/**
 * Challenge 04 : Sum first 1000 primes
 * Created by cschalk on 3/21/16.
 */
public class Challenge04 {

    private static int totalSum = 0;
    private static int primesFound = 0;

    public static void main(String[] args) {
        int test = 1;
        do {
            if (isPrime(test)) {
                totalSum += test;
                primesFound++;
            }
            test++;
        } while (primesFound < 1000);
        System.out.println(totalSum);
        System.exit(1);
    }

    private static boolean isPrime(int in) {
        if (in <= 1) { return false; }
        if (in == 2) { return true; }
        for (int i = 2 ; i < Math.sqrt(in) + 1 ; i++ ) {
            if (in % i == 0) { return false; }
        }
        return true;
    }

}

File: code_eval/cpp/easy/challenge_04.cpp
// Created by cschalk on 3/21/16.
// Challenge 04 : Sum of first 1000 primes
#include <iostream>
#include <math.h>

int sum = 2;
int primes = 1;

bool isPrime(int test) {
    switch (test) {
        case 1  : return false;
        case 2  : return true;
        default : {
            if (test % 2 == 0) {
                return false;
            } else {
                auto root = sqrt(test) + 1;
                for (auto i = 3; i < root; i++) {
                    if (test % i == 0) return false;
                }
                return true;
            }
        }
    }
}

int main(int argc, char **argv) {
    int number = 3;
    do {
        if (isPrime(number)) {
            sum += number;
            primes++;
        }
        number+=2;
    } while (primes < 1000);

    std::cout << sum << std::endl;
    return 0;
}